src/xb-node.c

```diff
@@ -365,6 +389,70 @@ xb_node_get_attr_as_uint (XbNode *self, const gchar *name)
 	return g_ascii_strtoull (tmp, NULL, 10);
 }
 
+/**
+ * xb_node_attr_iter_init:
+ * @iter: an uninitialized #XbNodeIter
+ * @self: a #XbNode
+ *
+ * Initializes a name/value pair iterator for the node attributes
+ * and associates it with @self.
+ *
+ * Since: 0.3.4
+ **/
+void
+xb_node_attr_iter_init (XbNodeIter *iter, XbNode *self)
+{
+	XbNodePrivate *priv = GET_PRIVATE (self);
+	RealIter *ri = (RealIter *) iter;
+
+	g_return_if_fail (iter != NULL);
+	g_return_if_fail (XB_IS_NODE (self));
+
+	ri->node = self;
+	ri->position = xb_silo_node_get_attr_count (priv->sn);
+}
+
+/**
+ * xb_node_attr_iter_next:
+ * @iter: an initialized #XbNodeIter
+ * @name: (out): Destination of the returned attribute name
+ * @value: (out): Destination of the returned attribute value
+ *
+ * Returns the current attribute name and value and advances the iterator.
```

Comment:
It would be helpful to include a brief code example of how to do a loop with `xb_node_attr_iter_{init,next}()`, which people can just copy and adapt into their code.

e.g.
```
 * |[
 * XbNodeAttrIter iter;
 * const gchar *attr_name, *attr_value;
 *
 * xb_node_attr_iter_init (&iter, node);
 * while (xb_node_attr_iter_next (&iter, &attr_name, &attr_value))
 *   {
 *     /* use attr_name and attr_value; no need to free them */
 *   }
 * ]|
```